client/src/components/overview/overview-components/Carousel.jsx

```diff
@@ -1,22 +1,157 @@
+  cursor: pointer;
+  transition: 0.5s;
+  color: transparent;
+  &:hover {
+    background-color: rgba(255,255,255,0.3);
+    border-radius: 10px;
+    color: black;
+  }
+`
+// right button for carousel
+const RightButton = styled.button`
+  position: absolute;
+  right: 1%;
+  width: 10%;
+  height: 60%;
+  border: none;
+  background-color: transparent;
+  color: black;
+  cursor: pointer;
+  transition: 0.5s;
+  color: transparent;
+  &:hover {
+    background-color: rgba(255,255,255,0.3);
+    border-radius: 10px;
+    color: black;
+  }
+`
+
 export default function OverviewCarousel({ photos }) {
-  var photoList = photos.map((photo) => {
-    return <div key={photo.url}><img src={photo.url}/></div>
-  })
+  // this state is the viewpoint that the carousel is currently at, when the button is pressed, it increases the current viewpoint of the div by 100, thus changing the photo.
+  // Here is a link to the carousel video I watched to get this down. Bad music but good video
+    // https://www.youtube.com/watch?v=Tdpq-9XYoNM
+  var [x, setx] = useState(0)
+  // on click function to move the carousel to the left
+  const goLeft = () => {
+    setx(x + 100);
+  }
+  // on click function to move the carousel to the right
+  const goRight = () => {
+    setx(x - 100);
+  }
+
   return (
   <StyledOverviewCarousel>
-    <Carousel>
-      {photoList}
-    </Carousel>
+    {
+      photos.map((photo, index) => {
+        return (
+          // This translate x transformation is given to the slide div because it allows the picture to be shown that correlates with the x axis vertex
+          // i.e. the first picture is at x=0, the second is x=100, the third is x=200
+          // updating the x value changes what picture is shown
+          <Slide key={index} style={{transform: `translateX(${x}%)`}}>
+            <img src={photo.url}/>
+          </Slide>
+        )
+      })
+    }
+    <ThumbnailSelector>
+      {
+        photos.map((photo, index) => {
+          // if the thumbnail is selected, change the opacity of the image
+          if (x === -100*index) {
+            return (
+              <Thumbnail key={index} style={{opacity: '0.5'}}onClick={() => {
+                setx(x = -100*index)
+              }}>
+                <img src={photo.thumbnail_url}></img>
+              </Thumbnail>
+            )
+          } else {
+            return (
+              <Thumbnail key={index} onClick={() => {
+                setx(x = -100*index)
+              }}>
+                <img src={photo.thumbnail_url}></img>
+              </Thumbnail>
+            )
+          }
+        })
+      }
+    </ThumbnailSelector>
+    {/* these two ternary statements are to render in either a button or nothing depending on what position the gallery is in */}
+    {x === 0 ? (<div></div>) : (<LeftButton onClick={goLeft}>left</LeftButton>)}
+    {x > (-100 * photos.length + 100) ? (<RightButton onClick={goRight}>right</RightButton>) : (<div></div>)}
```

Comment:
wont x always be greater that -100 * photos.length +100 if the length is at least 2 since x will always be from 0 to photos.length * 100 - 100?

Reply:
The further down the carousel you get, the more negative the number becomes. I didn't put explain this correctly in the earlier comment that I made. X becomes smaller (a larger negative) as you click to the right. This function stops you when the negative number becomes larger than the array length * -100.